### PR TITLE
Fixed ov_coverage target

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Added `ov_coverage` target, `ie_coverage` exists for backward compatibility
 - Removed `openvino_all` target for coverage
 - Skip incorrect info files
 - Enable code coverage for macOS

### Tickets:
 - NA

### CODE REVIEW

File: cmake/developer_package/coverage/coverage.cmake [hunk 1]
```diff
@@ -4,210 +4,231 @@
...
+# ov_coverage_genhtml(INFO_FILE <info_file> PREFIX <prefix>)
+#
+function(ov_coverage_genhtml)
+    cmake_parse_arguments(OV_COVERAGE "" "INFO_FILE;PREFIX" "" ${ARGN})
+
+    set(input_file "${OV_COVERAGE_REPORTS}/${OV_COVERAGE_INFO_FILE}.info")
+    set(output_directory "${OV_COVERAGE_REPORTS}/${OV_COVERAGE_INFO_FILE}")
+
+    add_custom_command(OUTPUT "${output_directory}/index.html"
+                       COMMAND [ -s ${input_file}  ] && genhtml ${input_file} --title "${OV_COVERAGE_INFO_FILE}" --legend
```
Comment (ilyachur) [inline cmake/developer_package/coverage/coverage.cmake:156]: `[ -s ${input_file}  ] && ... || echo "Skip ${input_file}"` is a fix for empty info files

File: cmake/coverage.cmake [hunk 2]
```diff
@@ -4,95 +4,102 @@
...
-ie_coverage_genhtml(INFO_FILE "openvino_all"
-                    PREFIX "${OV_COVERAGE_BASE_DIRECTORY}")
-
 if(ENABLE_OV_ONNX_FRONTEND)
-    ie_coverage_extract(INPUT "openvino" OUTPUT "onnx"
+    ov_coverage_extract(INPUT "openvino" OUTPUT "onnx"
         PATTERNS "${OV_COVERAGE_BASE_DIRECTORY}/src/frontends/onnx/*"
         "${OV_COVERAGE_BASE_DIRECTORY}/src/frontends/onnx/*")
-    ie_coverage_genhtml(INFO_FILE "onnx"
+    ov_coverage_genhtml(INFO_FILE "onnx"
```
Comment (ilya-lavrenov) [inline cmake/coverage.cmake:103]: can we add other FEs?

File: cmake/coverage.cmake [hunk 3]
```diff
@@ -4,95 +4,102 @@
...
+ov_coverage_extract(INPUT "openvino" OUTPUT "legacy"
+                    PATTERNS "${OV_COVERAGE_BASE_DIRECTORY}/src/common/legacy/*")
+ov_coverage_genhtml(INFO_FILE "legacy"
                     PREFIX "${OV_COVERAGE_BASE_DIRECTORY}")
 
-ie_coverage_extract(INPUT "openvino" OUTPUT "transformations"
-                    PATTERNS "${OV_COVERAGE_BASE_DIRECTORY}/src/common/transformations/*")
-ie_coverage_genhtml(INFO_FILE "transformations"
-                    PREFIX "${OV_COVERAGE_BASE_DIRECTORY}")
+if(ENABLE_INTEL_CPU)
```
Comment (ilya-lavrenov) [inline cmake/coverage.cmake:47]: why hetero under CPU?

Let's make sure all the components are covered here

File: cmake/coverage.cmake (hunk 2 above)
Comment (ilyachur) [reply cmake/coverage.cmake]: Fixed

File: cmake/coverage.cmake (hunk 3 above)
Comment (ilyachur) [reply cmake/coverage.cmake]: Typo fixed

File: cmake/coverage.cmake [hunk 4]
```diff
@@ -4,95 +4,149 @@
...
 
-ie_coverage_clean(REPOSITORY "openvino"
+ov_coverage_clean(REPOSITORY "openvino"
                   DIRECTORY "${OV_COVERAGE_GCDA_DATA_DIRECTORY}")
-ie_coverage_capture(INFO_FILE "openvino"
+ov_coverage_capture(INFO_FILE "openvino"
                     BASE_DIRECTORY "${OV_COVERAGE_BASE_DIRECTORY}"
-                    DIRECTORY "${OV_COVERAGE_GCDA_DATA_DIRECTORY}")
+                    DIRECTORY "${OV_COVERAGE_GCDA_DATA_DIRECTORY}"
+                    PATTERNS "${OV_COVERAGE_BASE_DIRECTORY}/*.pb.cc"
```
Comment (ilya-lavrenov) [inline cmake/coverage.cmake:12]: `EXCLUDE_PATTERNS`

File: cmake/developer_package/features.cmake [hunk 5]
```diff
@@ -33,7 +33,7 @@ ie_option (ENABLE_UB_SANITIZER "enable UndefinedBahavior sanitizer" OFF)
 
 ie_option (ENABLE_THREAD_SANITIZER "enable checking data races via ThreadSanitizer" OFF)
 
-ie_dependent_option (ENABLE_COVERAGE "enable code coverage" OFF "CMAKE_CXX_COMPILER_ID STREQUAL GNU" OFF)
+ie_dependent_option (ENABLE_COVERAGE "enable code coverage" OFF "CMAKE_CXX_COMPILER_ID STREQUAL GNU OR APPLE" OFF)
```
Comment (ilya-lavrenov) [inline cmake/developer_package/features.cmake:36]: `OR OV_COMPILER_IS_CLANG`

**Pull request merged**